### PR TITLE
Semantic scene ID assigned by default

### PR DESCRIPTION
**Pull request by aclegg3**
## Motivation and Context

Related to issue #284. A semantic scene ID should be assigned by default so that semantic rendering can be done regardless of the provided annotations. For example on the empty "NONE" scene.

In the future we would like to provide a method for setting those semantic annotations for physics objects programmatically.

Comments welcome.

## How Has This Been Tested

Local testing.

Empty scene with physics object rendered via rgb and semantic sensors:
image
(Image courtesy of a script by Daniel Huber.)

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/esp/gfx/Simulator.cpp [hunk 1]
```diff
@@ -101,6 +101,12 @@ void Simulator::reconfigure(const SimulatorConfiguration& cfg) {
       throw std::invalid_argument("Cannot load: " + sceneFilename);
     }
 
+    activeSemanticSceneID_ = activeSceneID_;
```
Comment (erikwijmans) [inline src/esp/gfx/Simulator.cpp]: This is the only line that is needed.  And 132 to 136 can go.

**Comment by aclegg3** [comment]
> This maybe somewhat confusing for people as it'll seem like instance renderings for datasets without them (i.e. Gibson) exist.

The primary use case of this is programmatic assignment of semantics to physics objects and loading those objects into arbitrary or empty scenes. Some use cases may not care about structural semantics and only those for imported objects. 

Maybe we should for now just add the empty scene as a criterion for doing this along with:
```
if (sceneInfo.type == assets::AssetType::SUNCG_SCENE ||
        sceneInfo.type == assets::AssetType::INSTANCE_MESH) 
{ 
activeSemanticSceneID_ = activeSceneID_;
}
```

**Comment by aclegg3** [comment]
> This maybe somewhat confusing for people as it'll seem like instance renderings for datasets without them (i.e. Gibson) exist.

Ok, now this only affects the EMPTY_SCENE.

**Comment by erikwijmans** [comment]
Making it just for the empty scene seems like a band-aid that we will just have to revisit in a couple months when someone wants a scene in the background but wants segmentation of added objects.  Probably enable it for all scenes but add a comment about this and add a logging statement when a scene is loaded without an instance-segmentation mesh?

File: src/esp/gfx/Simulator.cpp [hunk 2]
```diff
@@ -125,7 +125,8 @@ void Simulator::reconfigure(const SimulatorConfiguration& cfg) {
 
     // instance meshes and suncg houses contain their semantic annotations
     if (sceneInfo.type == assets::AssetType::SUNCG_SCENE ||
-        sceneInfo.type == assets::AssetType::INSTANCE_MESH) {
+        sceneInfo.type == assets::AssetType::INSTANCE_MESH ||
+        sceneFilename.compare(assets::EMPTY_SCENE) == 0) {
```
Comment (mathfac) [inline src/esp/gfx/Simulator.cpp]: Let's add TODO to create mesh programmatically in future.

File: src/esp/gfx/Simulator.cpp [hunk 3]
```diff
@@ -121,13 +121,17 @@ void Simulator::reconfigure(const SimulatorConfiguration& cfg) {
...
-        sceneInfo.type == assets::AssetType::INSTANCE_MESH ||
-        sceneFilename.compare(assets::EMPTY_SCENE) == 0) {
+    }else{
       activeSemanticSceneID_ = activeSceneID_;
+      // instance meshes and suncg houses contain their semantic annotations
+      // empty scene has none to worry about
+      if ( !(sceneInfo.type == assets::AssetType::SUNCG_SCENE ||
+        sceneInfo.type == assets::AssetType::INSTANCE_MESH ||
+        sceneFilename.compare(assets::EMPTY_SCENE) == 0)) {
+          //TODO: programmatic generation of semantic meshes when no annotiations are provided.
```
Comment (erikwijmans) [inline src/esp/gfx/Simulator.cpp]: > //TODO: programmatic generation of semantic meshes when no annotiations are provided.

That sounds like a massive research project, not a TODO

**Pull request merged**